src/Entity/ItemId.php

```diff
@@ -24,7 +24,7 @@ class ItemId extends EntityId implements Int32EntityId {
 	 */
 	public function __construct( $idSerialization ) {
 		$this->assertValidIdFormat( $idSerialization );
-		$this->serialization = strtoupper( $idSerialization );
+		parent::__construct( parent::normalizeIdSerialization( $idSerialization ) );
```

Comment:
The idea was to do normalization in the constructor and to make `serialization` private to the parent class which isn't possible yet since it's still used in the `unserialize` methods of the subclasses.